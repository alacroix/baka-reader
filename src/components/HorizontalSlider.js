// @flow

import React from 'react';
import { StyleSheet, ScrollView, View } from 'react-native';
import appStyle from 'bakareader/src/appStyle';

const styles = StyleSheet.create({
  wrapper: {
    marginTop: appStyle.grid.x4,
    marginBottom: appStyle.grid.x2,
  },
  content: {
    paddingLeft: appStyle.grid.x2,
    paddingRight: appStyle.grid.x2,
  },
});

type PropsType = {
  children: Array<any>
}

function HorizontalSlider(props: PropsType) {
  return (
    <View style={styles.wrapper}>
      <ScrollView
        horizontal
        contentContainerStyle={styles.content}
        showsHorizontalScrollIndicator={false}
      >
        {props.children}
      </ScrollView>
    </View>
  );
}

export default HorizontalSlider;
