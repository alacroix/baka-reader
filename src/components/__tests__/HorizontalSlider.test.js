import { Text } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import HorizontalSlider from '../HorizontalSlider';

describe('HorizontalSlider', () => {
  it('should render the horizontal slider', () => {
    const tree = renderer.create(
      <HorizontalSlider>
        <Text>Test</Text>
        <Text>Test 2</Text>
      </HorizontalSlider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
