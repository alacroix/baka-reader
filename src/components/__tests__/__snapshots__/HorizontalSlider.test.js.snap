// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalSlider should render the horizontal slider 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
      "marginTop": 32,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Test
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Test 2
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
